osu.Game/Overlays/Profile/Header/Components/FollowersButton.cs

```diff
@@ -13,15 +21,192 @@ public partial class FollowersButton : ProfileHeaderStatisticsButton
     {
         public readonly Bindable<UserProfileData?> User = new Bindable<UserProfileData?>();
 
-        public override LocalisableString TooltipText => FriendsStrings.ButtonsDisabled;
+        // Because it is impossible to update the number of friends after the operation,
+        // the number of friends obtained is stored and modified locally.
+        private int followerCount;
+
+        public override LocalisableString TooltipText
+        {
+            get
+            {
+                switch (status.Value)
+                {
+                    case FriendStatus.Self:
+                        return FriendsStrings.ButtonsDisabled;
+
+                    case FriendStatus.None:
+                        return FriendsStrings.ButtonsAdd;
+
+                    case FriendStatus.NotMutual:
+                    case FriendStatus.Mutual:
+                        return FriendsStrings.ButtonsRemove;
+                }
+
+                return FriendsStrings.TitleCompact;
+            }
+        }
 
         protected override IconUsage Icon => FontAwesome.Solid.User;
 
+        private readonly IBindableList<APIRelation> apiFriends = new BindableList<APIRelation>();
+        private readonly IBindable<APIUser> localUser = new Bindable<APIUser>();
+
+        private readonly Bindable<FriendStatus> status = new Bindable<FriendStatus>();
+
+        [Resolved]
+        private OsuColour colour { get; set; } = null!;
+
+        [Resolved]
+        private OverlayColourProvider colourProvider { get; set; } = null!;
+
         [BackgroundDependencyLoader]
-        private void load()
+        private void load(IAPIProvider api, INotificationOverlay? notifications)
+        {
+            localUser.BindTo(api.LocalUser);
+
+            status.BindValueChanged(_ =>
+            {
+                updateIcon();
+                updateColor();
+            });
+
+            User.BindValueChanged(u =>
+            {
+                followerCount = u.NewValue?.User.FollowerCount ?? 0;
+                updateStatus();
+            }, true);
+
+            apiFriends.BindTo(api.Friends);
+            apiFriends.BindCollectionChanged((_, _) => Schedule(updateStatus));
+
+            Action += () =>
+            {
+                if (User.Value == null)
+                    return;
+
+                if (status.Value == FriendStatus.Self)
+                    return;
+
+                ShowLoadingLayer();
+
+                APIRequest req = status.Value == FriendStatus.None ? new FriendAddRequest(User.Value.User.OnlineID) : new FriendDeleteRequest(User.Value.User.OnlineID);
+
+                req.Success += () =>
+                {
+                    followerCount += status.Value == FriendStatus.None ? 1 : -1;
+
+                    api.UpdateLocalFriends();
```

Comment:
Does this really need to refetch friends every time? The request just succeeded, you can tell what happened, why do a refetch?

I guess it would make sense in the case of adding users, because you might not have the `APIUser` instance present or whatever, but isn't/can't that be in the add friend request? Or even couldn't the `APIUser` be taken from the ambient overlay?

Reply:
I tried removing the refetch but things crash, something or other about groups not being populated in the response of this.

@ppy/team-client can I get a second pair of eyes on this to make sure you're ok with it?